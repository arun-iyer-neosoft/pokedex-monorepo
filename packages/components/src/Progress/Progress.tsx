import React from 'react';

interface ProgressProps {
	max: number;
	base: number;
	background?: string;
	gradient?: boolean;
	fromColor?: string;
	toColor?: string;
	textColor?: string;
}

export const Progress = ({
	max,
	base,
	background,
	fromColor,
	gradient,
	toColor,
	textColor,
}: ProgressProps) => {
	let backgroundColor = background ?? 'bg-blue-600';
	if (gradient) {
		backgroundColor = `bg-gradient-to-r ${fromColor ?? 'from-purple-500'} ${
			toColor ?? 'to-fuchsia-500'
		}`;
	}
	let text = textColor ?? 'text-white';
	const width = (base * 100) / max;
	return (
		<div
			className={`h-4 w-full rounded-sm ${backgroundColor} ${text}`}
			style={{ width: `${width}%` }}
			data-testid='progress-bar'
		></div>
	);
};
