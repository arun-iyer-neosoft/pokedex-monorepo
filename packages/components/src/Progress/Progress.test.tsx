import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { Progress } from './Progress';

describe('Progress', () => {
	it('Renders the Progress Bar', () => {
		const { container } = render(<Progress max={100} base={20} />);

		expect(screen.getByTestId('progress-bar')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('Renders the Progress Bar with Width', () => {
		const { container } = render(<Progress max={100} base={20} />);

		expect(screen.getByTestId('progress-bar')).toHaveStyle('width:20%;');
		expect(container).toMatchSnapshot();
	});

	it('Renders the Progress Bar with background color', () => {
		const { container } = render(
			<Progress max={100} base={20} background='bg-red-400' />
		);

		expect(screen.getByTestId('progress-bar')).toHaveClass('bg-red-400');
		expect(container).toMatchSnapshot();
	});

	it('Renders the Progress Bar with gradient', () => {
		const { container } = render(
			<Progress max={100} base={20} gradient={true} />
		);

		expect(screen.getByTestId('progress-bar')).toHaveClass('bg-gradient-to-r');
		expect(container).toMatchSnapshot();
	});
});
