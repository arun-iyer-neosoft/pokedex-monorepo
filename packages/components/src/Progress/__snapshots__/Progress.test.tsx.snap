// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Renders the Progress Bar 1`] = `
<div>
  <div
    class="h-4 w-full rounded-sm bg-blue-600 text-white"
    data-testid="progress-bar"
    style="width: 20%;"
  />
</div>
`;

exports[`Progress Renders the Progress Bar with Width 1`] = `
<div>
  <div
    class="h-4 w-full rounded-sm bg-blue-600 text-white"
    data-testid="progress-bar"
    style="width: 20%;"
  />
</div>
`;

exports[`Progress Renders the Progress Bar with background color 1`] = `
<div>
  <div
    class="h-4 w-full rounded-sm bg-red-400 text-white"
    data-testid="progress-bar"
    style="width: 20%;"
  />
</div>
`;

exports[`Progress Renders the Progress Bar with gradient 1`] = `
<div>
  <div
    class="h-4 w-full rounded-sm bg-gradient-to-r from-purple-500 to-fuchsia-500 text-white"
    data-testid="progress-bar"
    style="width: 20%;"
  />
</div>
`;
